### ISSUE TITLE
Is it possible to run DEX behing Apache Proxy?

### ISSUE THREAD

**Issue by bviviano** [OSC/ondemand#2174]
Hello.  I have a configuration question.  Is it currently possible to run DEX behind Apache, instead of directly on port 5554?

So instead of `https://myhost.example.gov:5554/` it could be `https://myhost.example.gov/dex/`?

I see from this ticket

https://github.com/OSC/ondemand/issues/1303

It was discussed a bit, but maybe isn't currently supported.

I have ondemand-dex running with LDAP on 5554 currently and it's working as expected, but my security group freaks out anytime they see web traffic not on an expected Web Port (80 or 443) when they run their regular system scans, so to avoid this I am trying to see if there is a way to "hide" port 5554 behind a ReverseProxy, inside the current Apache server running on 443.

Thanks.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201735133575781/1202682008197773) by [Unito](https://www.unito.io)

**Comment by johrstrom** [comment]
It seems possible but we don't have it automated. @treydock could they just create another vhost for dex?

**Comment by johrstrom** [comment]
It doesn't seem like this is possible without editing our `conf` file and probably even the dex configs too.  Seems like a patch may come in 2.1 though.

**Comment by bviviano** [comment]
Do you have any suggestions as to what to tweak/adjust if I wanted to run/test now with OOD 2.0?

I understand that making changes directly to /etc/httpd/conf.d/ood-portal.conf and/or /etc/ood/dex/config.yaml could have unintended consequences if update_ood_portal is re-run, but I'm willing to give it a go on my system to test, if you can suggest the changes needed.

I've done other setups with proxy/reverseproxy in Apache (Gitea, Grafana, etc), but between OOD and DEX there are a lot of moving parts I'm not 100% comfortable with getting all the rules correct.

We're using xCAT as our cluster manager and I've written a postscript to setup the OOD portal on boot, for the node(s) that will run OOD, so adding some tweaks via sed or similar to adjust ood-portal.conf / config.yaml after the postscript runs update_ood_portal wouldn't be an issue, until the 2.1 release that could have the settings directly in /etc/ood/config/ood_portal.yml.

Thanks.

**Comment by treydock** [comment]
So right now the only way to support this is direct edits for /etc/httpd/conf.d/ood-portal.conf and /etc/ood/dex/config.yaml.

For ood-portal.conf:

Add this virtualhost for OnDemand:

```
  ProxyRequests Off
  ProxyPreserveHost On
  ProxyAddHeaders On
  ProxyPass /dex http://localhost:5556/dex
  ProxyPassReverse /dex http://localhost:5556/dex
```

Edit `OIDCProviderMetadataURL` to point at ServerName of your OnDemand host and remove port 5556 and ensure the path begins with `/dex`.

For dex/config.yam:

Edit `issuer` to remove port , ensure same hostname as OnDemand and add `/dex` path.

For `web` remove `https`.  For `http` change to `localhost:5556`.

The OnDemand 2.0.28 release which is coming soon will support this by simply doing this in ood_portal.yaml:

```yaml
dex_uri: /dex
```

**Comment by johrstrom** [comment]
It'll be in 2.0. That should be in `latest` later this week and available for the public sometime next week. Trey seems to have the instructions to hot patch your system immediately.

**Comment by bviviano** [comment]
@treydock / @johrstrom If it'll be available in 2.0.28 in a week or so, I can wait.  The ticket I saw said 2.1, which could be a while.  I'll keep an eye out for the 2.0.28 release and give the proxy setup a test.  Thanks.
